bio-submission.txt

```diff
@@ -0,0 +1,41 @@
+Bio submission overview
+=======================
+
+Since btrfs has support for various raid levels it's submission path is more
+involved than a traditional filesystem. For the purposes of this document a
+raid1-based filesystem is going to be considered.
+
+
+Submission
+----------
+
+The upper layers of the filesystem would create a single bio for a particular
+logical address. However, due to the way btrfs manages its space the first
+thing that needs to happen is map this logical address to physical one.
+Additionally, depending on the raid profile of the portion of disk where the
+requested data falls in it's possible that the single io is actually going to
+turn into multiple io requestes.
+
+This mapping occurs in btrfs_map_bio. First a btrfs_bio struct is created.  It
+holds information about the underlying properties of the range that we want to
+interact with. This is information such as how many stripes (i.e copies of the
+data there are on disk), length of the stripe and copies of some value of the
+initial bio which are going to be replaced. Then for every stripe a bio is
```

Comment:
"...going to be replaced in the process of remapping the request to the appropriate underlying devices".